### PR TITLE
Use CoordsXY for for some types

### PR DESCRIPTION
**Pull request by ZehMatt**
The next refactor should be probably converting Peep x and y into CoordsXY. 

NOTE: Replays require an update, CoordsXY uses 32 bit instead of 16, so the checksums will mismatch.

### CODE REVIEW

File: src/openrct2/peep/GuestPathfinding.cpp [hunk 1]
```diff
@@ -158,18 +157,18 @@ static int32_t peep_move_one_tile(Direction direction, Peep* peep)
         if (direction == 0 || direction == 2)
         {
             // Peep is moving along X, so apply the offset to the X position of the destination and clamp their current Y
-            peep->DestinationX += offset;
+            peep->Destination.x += offset;
             const uint16_t centerLine = (peep->y & 0xFFE0) + COORDS_XY_HALF_TILE;
-            peep->DestinationY = std::clamp(
+            peep->Destination.y = std::clamp(
                 peep->y, static_cast<int16_t>(centerLine - 3), static_cast<int16_t>(centerLine + 3));
```
Comment (Gymnasiast) [inline src/openrct2/peep/GuestPathfinding.cpp:163]: I think you can make everything int32_t here and do away with the casting.

**Comment by ZehMatt** [comment]
I was thinking I should also convert more of {x, y} to CoordsXY otherwise I have to update the replays over and over.

File: src/openrct2/peep/GuestPathfinding.cpp (hunk 1 above)
Comment (tupaschoal) [reply src/openrct2/peep/GuestPathfinding.cpp]: Some of these places change how the game work if we change them to 32 bits, gotta be careful (I don't remember which ones though, just faced this once when I was migrating to CoordsXY(ZD))

**Comment by ZehMatt** [comment]
I don't like the way I did it. I should have a getter and setter and leave the type alone for the time being.

**Comment by ZehMatt** [comment]
I managed to avoid breaking replays by first implementing getters and setters, this makes the code a bit more readable. This is a good starting point for more refactors.

File: src/openrct2/peep/Staff.cpp [hunk 2]
```diff
@@ -653,9 +653,7 @@ bool Staff::DoHandymanPathFinding()
     }
 
     PeepDirection = newDirection;
-    DestinationX = chosenTile.x + 16;
-    DestinationY = chosenTile.y + 16;
-    DestinationTolerance = 3;
+    SetDestination(chosenTile + CoordsXY{ 16, 16 }, 3);
```
Comment (duncanspumpkin) [inline src/openrct2/peep/Staff.cpp:656]: Can we make these toTileCentre?

File: src/openrct2/peep/Staff.cpp [hunk 3]
```diff
@@ -1173,8 +1169,8 @@ void Staff::UpdateMowing()
             return;
         }
 
-        DestinationX = _MowingWaypoints[Var37].x + NextLoc.x;
-        DestinationY = _MowingWaypoints[Var37].y + NextLoc.y;
+        auto destination = _MowingWaypoints[Var37] + NextLoc;
+        SetDestination(destination);
```
Comment (duncanspumpkin) [inline src/openrct2/peep/Staff.cpp:1173]: Assume this will be tolerance 3 as that's what it was too get to the first waypoint. Maybe do a follow on pr that forces setting the tolerance.

File: src/openrct2/peep/Staff.cpp (hunk 2 above)
Comment (Gymnasiast) [reply src/openrct2/peep/Staff.cpp:656]: It's not guaranteed to be the same, as NextLoc is a CoordsXY as well, IIRC.

Comment (duncanspumpkin) [reply src/openrct2/peep/Staff.cpp:656]: We should go through at some point and either enforce using only to tile centre or clearly mark why it's not correct in this situation. Guess we can leave for later though.

File: src/openrct2/peep/Staff.cpp (hunk 3 above)
Comment (ZehMatt) [reply src/openrct2/peep/Staff.cpp:1173]: I initially did that where default was zero however that broke the replays as it changes the hash so for testing this was better.

**Review by duncanspumpkin — APPROVED** [review]

File: src/openrct2/peep/GuestPathfinding.cpp [hunk 4]
```diff
@@ -154,24 +151,24 @@ static int32_t peep_move_one_tile(Direction direction, Peep* peep)
...
-        int8_t offset = (scenario_rand() & 7) - 3;
+        const int8_t offset = (scenario_rand() & 7) - 3;
         if (direction == 0 || direction == 2)
         {
             // Peep is moving along X, so apply the offset to the X position of the destination and clamp their current Y
-            peep->DestinationX += offset;
-            const uint16_t centerLine = (peep->y & 0xFFE0) + COORDS_XY_HALF_TILE;
-            peep->DestinationY = std::clamp(
-                peep->y, static_cast<int16_t>(centerLine - 3), static_cast<int16_t>(centerLine + 3));
+            const int32_t centerLine = (peep->y & 0xFFE0) + COORDS_XY_HALF_TILE;
```
Comment (Gymnasiast) [inline src/openrct2/peep/GuestPathfinding.cpp:158]: `centreLine`

(We use British English throughout.)

File: src/openrct2/peep/Peep.cpp [hunk 5]
```diff
@@ -606,7 +606,8 @@ std::optional<CoordsXY> Peep::UpdateAction(int16_t& xy_distance)
         Action = PeepActionType::None2;
     }
 
-    CoordsXY diffrenceLoc = { x - DestinationX, y - DestinationY };
+    CoordsXY diffrenceLoc = GetLocation();
```
Comment (Gymnasiast) [inline src/openrct2/peep/Peep.cpp:609]: `differenceLoc`

File: src/openrct2/peep/Peep.cpp [hunk 6]
```diff
@@ -3381,3 +3375,20 @@ bool Peep::HasItem(ShopItem peepItem) const
...
+
+void Peep::SetDestination(const CoordsXY& coords, int32_t tolerance)
+{
+    SetDestination(coords);
+    DestinationTolerance = tolerance;
+}
+
+CoordsXY Peep::GetDestination() const
+{
+    return CoordsXY{ static_cast<int16_t>(DestinationX), static_cast<int16_t>(DestinationY) };
```
Comment (Gymnasiast) [inline src/openrct2/peep/Peep.cpp:3393]: Why did you add these casts?

File: src/openrct2/peep/Staff.cpp [hunk 7]
```diff
@@ -1449,17 +1445,15 @@ void Staff::UpdateHeadingToInspect()
...
+        SetDestination({ destX, destY }, 2);
         sprite_direction = PeepDirection << 3;
 
         z = rideEntranceExitElement->base_height * 4;
         SubState = 4;
         // Falls through into SubState 4
     }
 
-    int16_t delta_y = abs(y - DestinationY);
+    int16_t delta_y = abs(y - GetDestination().y);
```
Comment (Gymnasiast) [inline src/openrct2/peep/Staff.cpp:1456]: Looks like you're using y directly here instead of through `GetLocation()`.

File: src/openrct2/peep/Staff.cpp [hunk 8]
```diff
@@ -1449,17 +1445,15 @@ void Staff::UpdateHeadingToInspect()
         int32_t destX = NextLoc.x + 16 + DirectionOffsets[PeepDirection].x * 53;
```
Comment (Gymnasiast) [inline src/openrct2/peep/Staff.cpp:1445]: Should also be changed to use CoordsXY, but may be left for another PR (in which case I'd like you to create an issue for it.)

File: src/openrct2/world/Park.cpp [hunk 9]
```diff
@@ -722,10 +722,8 @@ Peep* Park::GenerateGuest()
             peep->sprite_direction = direction << 3;
 
             // Get the centre point of the tile the peep is on
```
Comment (Gymnasiast) [inline src/openrct2/world/Park.cpp:724]: This comment is now stating the obvious, so should be removed.

File: src/openrct2/peep/Peep.cpp (hunk 6 above)
Comment (ZehMatt) [reply src/openrct2/peep/Peep.cpp:3393]: DestinationX and DestinationY are uint16_t, so explicit cast. Those fields will be eventually superseded by CoordsXY.

Comment (Gymnasiast) [reply src/openrct2/peep/Peep.cpp:3393]: But uint16_t fits into int32_t, so why does it need to be cast?

Comment (ZehMatt) [reply src/openrct2/peep/Peep.cpp:3393]: Signed vs unsigned. But I think DestinationX and DestinationY never use negative numbers in that regard, I'll eliminate the casts.

**Review by Gymnasiast — APPROVED** [review]
